### PR TITLE
Replace Boost.Python with pybind11

### PR DESCRIPTION
**Pull request by valgur**
Hi! Thanks for the nice little library.

This PR:
* Replaces Boost.Python with a much, much more light-weight but functionally identical pybind11.
* Migrates to Python 3.
* Adds setup.py.
* Fixes some libavutil deprecations.

As a result, the `libh264decoder` Python package can be installed directly and conveniently with pip if the `libavcodec`, `libavutil` and `libswscale` libraries are installed.

```bash
pip install git+https://github.com/valgur/h264decoder.git
python -c 'import libh264decoder; print(dir(libh264decoder))'
```

### CODE REVIEW

**Comment by DaWelter** [comment]
Oh! Nice!

I'll take a look.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,25 +1,28 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.11)
```
Comment (DaWelter) [inline CMakeLists.txt]: Shouldn't we require 3.14? I'm looking at this guide https://cliutils.gitlab.io/modern-cmake/chapters/projects/fetch.html

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
+pkg_check_modules(libavcodec REQUIRED libavcodec)
 
-add_compile_options ("-std=c++0x")
+find_package(pybind11)
+if(pybind11_FOUND)
+  message("Using existing pybind11 v${pybind11_VERSION}")
+else()
+  message("Fetching latest pybind11")
+  include(FetchContent)
+  FetchContent_Declare(pybind11 GIT_REPOSITORY https://github.com/pybind/pybind11)
```
Comment (DaWelter) [inline CMakeLists.txt]: For me this fetches the dev branch. 
```suggestion
  FetchContent_Declare(
    pybind11 
    GIT_REPOSITORY https://github.com/pybind/pybind11
    GIT_TAG v2.5)
```
Would be safer to fetch a release version.

File: h264decoder.cpp [hunk 3]
```diff
@@ -113,13 +127,13 @@ const AVFrame& ConverterRGB24::convert(const AVFrame &frame, ubyte* out_rgb)
...
                                  w, h, (AVPixelFormat)pix_fmt, 
-                                 w, h, PIX_FMT_RGB24, SWS_BILINEAR, 
+                                 w, h, AV_PIX_FMT_RGB24, SWS_BILINEAR,
                                  nullptr, nullptr, nullptr);
   if (!context)
     throw H264DecodeFailure("cannot allocate context");
   
   // Setup framergb with out_rgb as external buffer. Also say that we want RGB24 output.
-  avpicture_fill((AVPicture*)framergb, out_rgb, PIX_FMT_RGB24, w, h);
+  av_image_fill_arrays(framergb->data, framergb->linesize, out_rgb, AV_PIX_FMT_RGB24, w, h, 1);
```
Comment (DaWelter) [inline h264decoder.cpp:136]: No version check and conditional compilation required here?

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        try:
+            out = subprocess.check_output(['cmake', '--version'])
+        except OSError:
+            raise RuntimeError("CMake must be installed to build the following extensions: " +
+                               ", ".join(e.name for e in self.extensions))
+
+        if platform.system() == "Windows":
+            cmake_version = LooseVersion(re.search(r'version\s*([\d.]+)', out.decode()).group(1))
+            if cmake_version < '3.1.0':
+                raise RuntimeError("CMake >= 3.1.0 is required on Windows")
```
Comment (DaWelter) [inline setup.py]: This part is not correct, isn't it? We generally require 3.11 or 3.14. See my previous comment.

File: readme.md [hunk 5]
```diff
@@ -1,35 +1,34 @@
...
+
+* Other source files are tests and demos.
+
+
+Requirements
+------------
+* cmake for building
+* libav
+
+
```
Comment (DaWelter) [inline readme.md:29]: Could you please create a Credits section and add yourself. :-)

**Comment by DaWelter** [comment]
Hm ... guess I'll take my comments on myself unless you're still faster than me.

File: CMakeLists.txt (hunk 1 above)
Comment (valgur) [reply CMakeLists.txt]: Indeed, I updated the minimum required version.

File: h264decoder.cpp (hunk 3 above)
Comment (valgur) [reply h264decoder.cpp:136]: This works with the `libav` version released in 2016, which ships with Ubuntu 16.04. That seems like a reasonable amount of backwards-compatibility to me already, so I would not add an extra case here.

**Comment by valgur** [comment]
Everything should be addressed now. Thanks for the review and I appreciate the credits entry. :)

**Comment by DaWelter** [comment]
LGTM. I'm making a few tweaks for windows. See valgur-master branch. Then this shall become v2 of h264decoder.

**Comment by DaWelter** [comment]
Great work btw. Thanks a lot.

**Pull request merged**